test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -258,3 +278,64 @@ def test_multi_mom_hook_failure_affects_vnode(self):
                             id=self.moms.values()[0].shortname, expect=True)
         self.server.manager(MGR_CMD_CREATE, NODE,
                             id=self.moms.values()[0].shortname)
+
+    def test_multi_mom_hook_failure_affects_vnode2(self):
+        """
+        Run an execjob_begin hook that gets an exception
+        when executed by sister mom, causing
+        the server to execute the fail_action=offline_vnodes, which
+        result in sister vnode to be marked offline.
+        Then run an exechost_startup hook that gets an
+        exception when local mom is restarted. Vnode representing
+        local mom would be marked offline.
+        """
+        if len(self.moms.values()) != 2:
+            self.skipTest("Provide 2 moms while invoking test")
+
+        if self.moms.values()[0].is_cpuset_mom() is True or self.moms.values()[
+                1].is_cpuset_mom() is True:
+            self.skipTest("Skipping test on cpuset moms")
+
+        if self.is_cray is True:
+            self.skipTest("Skipping test on Crays")
+
+        # The moms provided to the test may have unwanted vnodedef files.
+        if self.moms.values()[0].has_vnode_defs():
+            self.moms.values()[0].delete_vnode_defs()
+        if self.moms.values()[1].has_vnode_defs():
+            self.moms.values()[1].delete_vnode_defs()
+
+        self.create_multi_vnodes(num_moms=2, num_vnode=1)
+
+        self.create_bad_begin_hook()
+
+        j1 = Job(TEST_USER)
+
+        j1.set_attributes({ATTR_l + '.select': '2:ncpus=1'})
+        j1.set_attributes({ATTR_l + '.place': 'scatter'})
+
+        jid = self.server.submit(j1)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid)
+
+        self.server.expect(NODE, {ATTR_NODE_state: 'free'},
+                           id=self.moms.values()[0].shortname, max_attempts=3,
+                           interval=2)
+        # sister mom's vnode gets offlined due to hook exception
+        self.server.expect(NODE,
+                           {ATTR_NODE_state: 'offline',
+                            ATTR_comment:
+                            "offlined by hook 'h2' due to hook error"},
+                           id=self.moms.values()[1].shortname,
+                           max_attempts=6, interval=2)

```

Comment:
To check for two attributes in an expect(), you need to add atterop=PTL_AND.  If you don't, it'll check one or the other.